core/views.py

```diff
@@ -112,3 +112,28 @@ class HackathonsRUDView(generics.RetrieveUpdateDestroyAPIView):
     serializer_class = HackathonSerializer
     lookup_field = 'pk'
     queryset = Hackathon.objects.all()
+
+class UserHackathonsView(generics.GenericAPIView):
+    """
+    API used to get the list of all the hackthons user registered for.
+    """
+    permission_classes = [HackathonPermissions]
+    
+
+class HackathonSubmissionView(generics.ListCreateAPIView):
+    """
+    API used to get the list of all the submissions of particular hackathon.
+    """
+    serializer_class = SubmissionsSerializer
+    permission_classes = [permissions.IsAdminUser]
```

Comment:
The GET request should have anonymous access so that we can show submissions on Hackathon page, while the POST request should have access to all users so that they can create the submission.

Reply:
Please take care of the fact that the results should be public only after the hackathon ends.

Reply:
@nishantwrp and the submissions too??

Reply:
Yup, a person can view only their submission before the submission deadline.